src/coreclr/tools/Common/TypeSystem/IL/DelegateInfo.cs

```diff
@@ -106,7 +106,6 @@ public TypeDesc Type
         public DelegateInfo(TypeDesc delegateType, DelegateFeature features)
         {
             Debug.Assert(delegateType.IsDelegate);
-            Debug.Assert(delegateType.IsTypeDefinition);
```

Comment:
I'm not sure if this is safe but NativeAOT otherwise asserts here.

Reply:
It isn't, the implementation is non-sensical for instantiated types.

Is this only used to get signature of the delegate? If so, replace this:

```
                DelegateInfo delegateInfo = _compilation.TypeSystemContext.GetDelegateInfo(calledType);
                int paramCountDelegateClosed = delegateInfo.Signature.Length + 1;
```

with this:

```
                int paramCountDelegateClosed = calledType.GetMethod("Invoke", null).Signature.Length + 1;
```